src/Build/Logging/BinaryLogger/BinaryLogger.cs

```diff
@@ -451,5 +451,69 @@ private void ProcessParameters(out bool omitInitialInfo)
                 throw new LoggerException(message, e, errorCode, helpKeyword);
             }
         }
+
+        private bool TryInterpretPathParameter(string parameter, out string filePath)
+        {
+            bool hasPathPrefix = parameter.StartsWith("LogFile=", StringComparison.OrdinalIgnoreCase);
+
+            bool isFileParam = hasPathPrefix || parameter.IndexOf('=') < 0;
+
+            if (!isFileParam)
+            {
+                filePath = null;
+                return false;
+            }
+
+            if (hasPathPrefix)
+            {
+                parameter = parameter.Substring("LogFile=".Length);
+            }
+
+            parameter = parameter.Trim('"');
+
+            bool isWildcard = ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_12) && parameter.Contains("{}");
+            bool hasProperExtension = parameter.EndsWith(".binlog", StringComparison.OrdinalIgnoreCase);
+            filePath = parameter;
+
+            if (!isWildcard)
+            {
+                return hasProperExtension;
+            }
+
+            filePath = parameter.Replace("{}", GetUniqueStamp(), StringComparison.Ordinal);
+
+            if (!hasProperExtension)
+            {
+                filePath += ".binlog";
+            }
```

Comment:
If the parameter presented but with incorrect extension, do we want to fail fast and report the error?

Reply:
In this case I'd preffer more friendly behavior - if the user forget to add the extension, it's automatically added.

I added it only for the case of wildcarded paths - not to break the general existing path... but I believe that in the short future we should have as general behavior (there is likely no good in throwing exception telling you you should add an extension)